webapp/components/features/FollowList/FollowList.spec.js

```diff
@@ -0,0 +1,150 @@
+        return {}
+      },
+      'auth/isModerator': () => false,
+    }
+  })
+
+  describe('mount', () => {
+    beforeAll(() => {
+      store = new Vuex.Store({
+        getters,
+      })
+    })
+
+    describe('given a type', () => {
+      describe('of `following`', () => {
+        it('uses the `following` data on :user', () => {
+          const wrapper = Wrapper({ user, type: 'following' })
+          wrapper.find('.base-button').trigger('click')
+
+          expect(wrapper.vm.allConnectionsCount).toBe(user.followingCount)
+          expect(wrapper.findAll('.user-teaser')).toHaveLength(user.following.length)
+          expect(wrapper.emitted('fetchAllConnections')).toEqual([['following']])
+        })
+      })
+
+      describe('of `followedBy`', () => {
+        it('uses the `followedBy` data on :user', () => {
+          const wrapper = Wrapper({ type: 'followedBy' })
+          wrapper.find('.base-button').trigger('click')
+
+          expect(wrapper.vm.allConnectionsCount).toBe(user.followedByCount)
+          expect(wrapper.findAll('.user-teaser')).toHaveLength(user.followedBy.length)
+          expect(wrapper.emitted('fetchAllConnections')).toEqual([['followedBy']])
+        })
+      })
+    })
+
+    describe('given no type', () => {
+      it('defaults type to `following`', () => {
+        const wrapper = Wrapper()
+        expect(wrapper.text()).toContain('profile.network.following')
+      })
+    })
+
+    describe('given a user', () => {
+      describe('without connections', () => {
+        it('displays the followingNobody message', () => {
+          const wrapper = Wrapper({ user: noConnectionsUser })
+          expect(wrapper.find('.nobody-message').text()).toContain(
+            'profile.network.followingNobody',
+          )
+        })
+
+        it('displays the followedByNobody message', () => {
+          const wrapper = Wrapper({ user: noConnectionsUser, type: 'followedBy' })
+          expect(wrapper.find('.nobody-message').text()).toContain(
+            'profile.network.followedByNobody',
+          )
+        })
+      })
+
+      describe('with up to 7 loaded connections', () => {
+        let wrapper
+        beforeAll(() => {
+          wrapper = Wrapper()
+        })
+
+        it(`renders the connections`, () => {
+          expect(wrapper.findAll('.user-teaser')).toHaveLength(user.following.length)
+        })
+
+        it(`has a button to load all remaining connections`, async () => {
+          wrapper.find('.base-button').trigger('click')
+          expect(wrapper.emitted('fetchAllConnections')).toBeTruthy()
+        })
+      })
+
+      describe('with more than 7 loaded connections', () => {
+        let wrapper
+        beforeAll(() => {
+          wrapper = Wrapper({ user: allConnectionsUser })
+        })
+
+        it('renders the connections', () => {
+          expect(wrapper.findAll('.user-teaser')).toHaveLength(allConnectionsUser.followingCount)
+        })
+
```

Comment:
> [<img alt="Mogge" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-04-15T14:33:08Z" title="Wednesday, April 15th 2020, 4:33:08 pm +02:00">Apr 15, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,150 @@
+import { config, mount } from '@vue/test-utils'
+import Vuex from 'vuex'
+
+import helpers from '~/storybook/helpers'
+import FollowList from './FollowList.vue'
+
+const localVue = global.localVue
+
+config.stubs['client-only'] = '<span><slot /></span>'
+config.stubs['ds-space'] = '<span><slot /></span>'
+config.stubs['nuxt-link'] = '<span><slot /></span>'
+
+const user = {
+  ...helpers.fakeUser()[0],
+  followedByCount: 12,
+  followingCount: 15,
+  followedBy: helpers.fakeUser(7),
+  following: helpers.fakeUser(7),
+}
+
+const allConnectionsUser = {
+  ...user,
+  followedBy: [
+    ...user.followedBy,
+    ...helpers.fakeUser(user.followedByCount - user.followedBy.length),
+  ],
+  following: [...user.following, ...helpers.fakeUser(user.followingCount - user.following.length)],
+}
+
+const noConnectionsUser = {
+  ...user,
+  followedByCount: 0,
+  followingCount: 0,
+  followedBy: [],
+  following: [],
+}
+
+describe('FollowList.vue', () => {
+  let store, getters
+  const Wrapper = (customProps) =>
+    mount(FollowList, {
+      store,
+      propsData: { user, ...customProps },
+      mocks: {
+        $t: jest.fn((str) => str),
+      },
+      localVue,
+    })
+
+  beforeAll(() => {
+    getters = {
+      'auth/user': () => {
+        return {}
+      },
+      'auth/isModerator': () => false,
+    }
+  })
+
+  describe('mount', () => {
+    beforeAll(() => {
+      store = new Vuex.Store({
+        getters,
+      })
+    })
+
+    describe('given a type', () => {
+      describe('of `following`', () => {
+        it('uses the `following` data on :user', () => {
+          const wrapper = Wrapper({ user, type: 'following' })
+          wrapper.find('.base-button').trigger('click')
+
+          expect(wrapper.vm.connections).toEqual(user.following)
+          expect(wrapper.vm.allConnectionsCount).toBe(user.followingCount)
+          expect(wrapper.emitted('fetchAllConnections')).toEqual([['following']])
+        })
+      })
+
+      describe('of `followedBy`', () => {
+        it('uses the `followedBy` data on :user', () => {
+          const wrapper = Wrapper({ type: 'followedBy' })
+          wrapper.find('.base-button').trigger('click')
+
+          expect(wrapper.vm.connections).toEqual(user.followedBy)
+          expect(wrapper.vm.allConnectionsCount).toBe(user.followedByCount)
+          expect(wrapper.emitted('fetchAllConnections')).toEqual([['followedBy']])
+        })
+      })
+    })
+
+    describe('given no type', () => {
+      it('defaults type to `following`', () => {
+        const wrapper = Wrapper()
+        expect(wrapper.vm.type).toBe('following')
+      })
+    })
+
+    describe('given a user', () => {
+      describe('without connections', () => {
+        it('displays the followingNobody message', () => {
+          const wrapper = Wrapper({ user: noConnectionsUser })
+          expect(wrapper.find('.nobody-message').text()).toBe(
+            `${noConnectionsUser.name} ${wrapper.vm.$t(`profile.network.followingNobody`)}`,
+          )
+        })
+
+        it('displays the followedByNobody message', () => {
+          const wrapper = Wrapper({ user: noConnectionsUser, type: 'followedBy' })
+          expect(wrapper.find('.nobody-message').text()).toBe(
+            `${noConnectionsUser.name} ${wrapper.vm.$t(`profile.network.followedByNobody`)}`,
+          )
+        })
+      })
+
+      describe('with up to 7 loaded connections', () => {
+        let wrapper
+        beforeAll(() => {
+          wrapper = Wrapper()
+        })
+
+        it(`renders the connections`, () => {
+          expect(wrapper.findAll('.user-teaser').length).toEqual(user.following.length)
+        })
+
+        it(`has a button to load all remaining connections`, async () => {
+          wrapper.find('.base-button').trigger('click')
+          expect(wrapper.emitted('fetchAllConnections')).toBeTruthy()
+        })
+      })
+
+      describe('with more than 7 loaded connections', () => {
+        let wrapper
+        beforeAll(() => {
+          wrapper = Wrapper({ user: allConnectionsUser })
+        })
+
+        it('renders the connections', () => {
+          expect(wrapper.findAll('.user-teaser')).toHaveLength(allConnectionsUser.followingCount)
+        })
+
```


Here is what I expected before. Looks good. But how does the `.--overflow` appear without triggering the click?

Reply:
> [<img alt="rbeer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/rbeer) **Authored by [rbeer](https://github.com/rbeer)**
_<time datetime="2020-04-15T20:37:17Z" title="Wednesday, April 15th 2020, 10:37:17 pm +02:00">Apr 15, 2020</time>_

---


The `connectionsClass` string is [computed](https://github.com/Human-Connection/Human-Connection/pull/3293/files#diff-68c74572b7e55ffbcef954e9c4d9d487R77) and then used in [line 7](https://github.com/Human-Connection/Human-Connection/pull/3293/files#diff-68c74572b7e55ffbcef954e9c4d9d487R7)